pkg/controller/vsphere/reconciler.go

```diff
@@ -115,6 +116,10 @@ func (r *Reconciler) update() error {
 		Ref:     vmRef,
 	}
 
+	if err := vm.reconcileTags(r.Context, r.session, r.machine); err != nil {
```

Comment:
I notice this is only done in the update, is it not possible to also tag the VM during create?

Reply:
I think we need clone task to finish before attaching the tag. I wasn't able to find another way.